### DISCUSSION TITLE
Add type annotations

### DISCUSSION THREAD

**Discussion by emiel** [maxcountryman/flask-login#846] [Ideas]
Thanks for flask-login. We still use it and we'd like to see type annotations added.

I could work on this if there is interest.

**Comment by lukasjuhrich** [comment]
Given that no maintainer has reacted for a few months: @emiel perhaps you might just create a PR? I'm also interested in having proper type hints. I could assist if required.

**Comment by davidism** [comment]
Based on my experience implementing type annotations in other libraries, I'm going to ask everyone to avoid making their own PRs for this. I can barely get it right, trying to review someone else modifying almost every line is impossible.

**Comment by emiel** [comment]
Thanks @lukasjuhrich for your message. I've switched jobs and am not using this library at this moment and so I won't be picking up this task myself.

@davidism: I trust your experience but someone needs to make a start on this if we want to make this happen. I'll leave it up to the maintainer on how to proceed... but we need an answer first I guess.

**Comment by kyzima-spb** [comment]
> Based on my experience implementing type annotations in other libraries, I'm going to ask everyone to avoid making their own PRs for this. I can barely get it right, trying to review someone else modifying almost every line is impossible.

IMHO, but annotations don't change the behavior in the code. If you have tests, you can run them and be sure that nothing is broken.

Perhaps in the future I would add them if there is support from the authors for the merge.

**Comment by davidism** [comment]
They don't but they do create a ton more noise when we don't get them exactly right and people start making issues about it that require more releases. I'll reiterate: please don't work on this unless you are literally a typing expert and will guarantee that you will keep responding to further typing issues for the next year after they're released.
